pkg/resources/external_volume.go

```diff
@@ -0,0 +1,717 @@
+			if ok && len(encryptionType) > 0 {
+				encryptionTypeParsed, err := sdk.ToS3EncryptionType(encryptionType)
+				if err != nil {
+					return nil, err
+				}
+
+				encryptionKmsKeyId, ok := storageLocationConfig["encryption_kms_key_id"].(string)
+				if ok && len(encryptionKmsKeyId) > 0 {
+					s3StorageLocation.Encryption = &sdk.ExternalVolumeS3Encryption{
+						Type:     encryptionTypeParsed,
+						KmsKeyId: &encryptionKmsKeyId,
+					}
+				} else {
+					s3StorageLocation.Encryption = &sdk.ExternalVolumeS3Encryption{
+						Type: encryptionTypeParsed,
+					}
+				}
+			}
+
+			storageLocation = sdk.ExternalVolumeStorageLocation{
+				S3StorageLocationParams: s3StorageLocation,
+			}
+		case sdk.StorageProviderGCS:
+			// Test that azure_tenant_id and storage_aws_role_arn are not given
+			// If given non empty plans will be produced
+			azureTenantId, ok := storageLocationConfig["azure_tenant_id"].(string)
+			if ok && len(azureTenantId) > 0 {
+				return nil, fmt.Errorf("unable to extract storage location, azure_tenant_id provided for gcs storage location")
+			}
+
+			storageAwsRoleArn, ok := storageLocationConfig["storage_aws_role_arn"].(string)
+			if ok && len(storageAwsRoleArn) > 0 {
+				return nil, fmt.Errorf("unable to extract storage location, storage_aws_role_arn provided for gcs storage location")
+			}
+
+			gcsStorageLocation := &sdk.GCSStorageLocationParams{
+				Name:           name,
+				StorageBaseUrl: storageBaseUrl,
+			}
+			encryptionType, ok := storageLocationConfig["encryption_type"].(string)
+			if ok && len(encryptionType) > 0 {
+				encryptionTypeParsed, err := sdk.ToGCSEncryptionType(encryptionType)
+				if err != nil {
+					return nil, err
+				}
+				encryptionKmsKeyId, ok := storageLocationConfig["encryption_kms_key_id"].(string)
+				if ok && len(encryptionKmsKeyId) > 0 {
+					gcsStorageLocation.Encryption = &sdk.ExternalVolumeGCSEncryption{
+						Type:     encryptionTypeParsed,
+						KmsKeyId: &encryptionKmsKeyId,
+					}
+				} else {
+					gcsStorageLocation.Encryption = &sdk.ExternalVolumeGCSEncryption{
+						Type: encryptionTypeParsed,
+					}
+				}
+			}
+
+			storageLocation = sdk.ExternalVolumeStorageLocation{
+				GCSStorageLocationParams: gcsStorageLocation,
+			}
+		case sdk.StorageProviderAzure:
+			// Test that storage_aws_role_arn and encryption_kms_key_id is not given
+			// If given non empty plans will be produced
+			storageAwsRolArn, ok := storageLocationConfig["storage_aws_role_arn"].(string)
+			if ok && len(storageAwsRolArn) > 0 {
+				return nil, fmt.Errorf("unable to extract storage location, storage_aws_role_arn provided for azure storage location")
+			}
+
+			encryptionKmsKeyId, ok := storageLocationConfig["encryption_kms_key_id"].(string)
+			if ok && len(encryptionKmsKeyId) > 0 {
+				return nil, fmt.Errorf("unable to extract storage location, encryption_kms_key_id provided for azure storage location")
+			}
+
+			// TODO
```

Comment:
Not sure exactly what's causing this issue, but possibly I’m being too strict trying to implement these checks - I can see there’s no equivalent in [storage_integration.go](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/pkg/resources/storage_integration.go#L171). I thought it might be useful to fail fast rather than let terraform produce non empty plans for no reason, for example when specifying an `azure_tenant_id` on an s3 storage location. But I don’t want to over complicate it if it’s not necessary - whatever your team thinks is best 👍 . If we do want to implement them, this issue should definitely be looked into before merging.

Reply:
This is now covered by https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3106#discussion_r1799069302